base/applications/cleanmgr/choicepage.c

```diff
@@ -0,0 +1,145 @@
+        lvC.cx = 158;
+        lvC.cchTextMax = 256;
+        lvC.fmt = LVCFMT_RIGHT;
+        
+        ListView_InsertColumn(hList, 0, &lvC);
+        ListView_InsertColumn(hList, 1, &lvC);
+
+        if (!CreateImageLists(hList))
+        {
+            MessageBoxW(NULL, L"CreateImageLists() failed!", L"Error", MB_OK | MB_ICONERROR);
+        }
+
+        StringCchPrintfW(TempList, _countof(TempList), L"%.02lf %s", SetOptimalSize(ChkDskSize), SetOptimalUnit(ChkDskSize));
+        AddItem(hList, L"Old ChkDsk Files", TempList, 0);
+        ZeroMemory(&TempList, sizeof(TempList));
+
+        StringCchPrintfW(TempList, _countof(TempList), L"%.02lf %s", SetOptimalSize(RappsSize), SetOptimalUnit(RappsSize));
+        AddItem(hList, L"RAPPS Files", TempList, 0);
+        ZeroMemory(&TempList, sizeof(TempList));
+        
+        StringCchPrintfW(TempList, _countof(TempList), L"%.02lf %s", SetOptimalSize(RecycleSize), SetOptimalUnit(RecycleSize));
+        AddItem(hList, L"Recycle Bin", TempList, 2);
+        ZeroMemory(&TempList, sizeof(TempList));
+        
+        if (bv.SysDrive == TRUE)
+        {
+            StringCchPrintfW(TempList, _countof(TempList), L"%.02lf %s", SetOptimalSize(TempSize), SetOptimalUnit(TempSize));
+            AddItem(hList, L"Temporary Files", TempList, 0);
+        }
+
+        LoadStringW(GetModuleHandleW(NULL), IDS_CLEANUP, StringText, _countof(StringText));
+        StringCchPrintfW(totalAmount, _countof(totalAmount), L"%.02lf %s", SetOptimalSize(TempSize + RecycleSize + ChkDskSize + RappsSize), SetOptimalUnit(TempSize + RecycleSize + ChkDskSize + RappsSize));
+        StringCchPrintfW(FullText, _countof(FullText), StringText, totalAmount, wcv.DriveLetter);
+
+        ZeroMemory(&seI, sizeof(seI));
+        seI.cbSize = sizeof seI;
+        seI.lpVerb = L"open";
+        seI.nShow = SW_SHOW;
+
+        SetDlgItemTextW(hwnd, IDC_STATIC_DLG, FullText);
+        ListView_SetCheckState(hList, 1, 1);
+        ListView_SetItemState(hList, 1, LVIS_SELECTED, LVIS_SELECTED);
+        return TRUE;
+
+    case WM_COMMAND:
+        switch (LOWORD(wParam))
+        {
+        case IDC_VIEW_FILES:
+            if (!PathIsDirectoryW(ViewFolder))
+            {
+                ZeroMemory(&StringText, sizeof(StringText));
+                LoadStringW(GetModuleHandleW(NULL), IDS_WARNING_FOLDER, StringText, _countof(StringText));
+                MessageBoxW(hwnd, StringText, L"Warning", MB_OK | MB_ICONWARNING);
+                break;
+            }
+
+            seI.lpFile = ViewFolder;
+            if (!ShellExecuteExW(&seI))
+            {
+                MessageBoxW(NULL, L"ShellExecuteExW() failed!", L"Ok", MB_OK | MB_ICONSTOP);
+            }
+            break;
+        }
+        break;
+
+    case WM_NOTIFY:
+    {
+        NMHDR* Header = (NMHDR*)lParam;
+        NMLISTVIEW* NmList = (NMLISTVIEW*)lParam;
+        LVITEMW lvI;
+        ZeroMemory(&lvI, sizeof(lvI));
+        static long long size = 0;
```

Comment:
that is not in any way a good idea btw.

Reply:
@learn-more So shall I just get the number out of the static control somehow and then calculate it instead of storing it in some static variable and keeping the count in it.

Reply:
Ideally you should store state in an object that you associate with the window, so that you can have multiple windows without them breaking in weird ways.

Reply:
@learn-more Uh I didn't get it.

Reply:
Note that this size variable is computed by that CheckedItem function, which in turn uses a global "bv" structure.
So I was even wondering whether that size couldn't be stored as well in that global structure?